e2e/chat.spec.ts

```diff
@@ -1,39 +1,127 @@
-import { test, expect } from '@playwright/test';
+import { describe, expect, beforeEach, getWalletId, test } from './fixtures';
 import { Chat } from './Chat';
+import { MetaMask } from './Metamask';
+import { EvmWalletID } from './Wallet';
 
-test('renders intro message', async ({ page }) => {
-  const chat = new Chat(page);
-  await chat.goto();
+describe('chat', () => {
+  test('renders intro message', async ({ page, context }) => {
+    const chat = new Chat(page);
+    await chat.goto();
 
-  expect(chat.messageContainer).not.toBeNull();
+    expect(chat.messageContainer).not.toBeNull();
 
-  await expect(page.getByText('Welcome to Oros!')).toBeVisible();
+    await expect(page.getByText("let's get started").first()).toBeVisible();
 
-  await expect(
-    page.getByText("I'm here to help you with all things blockchain and DeFi."),
-  ).toBeVisible();
-});
+    await expect(
+      page.getByText(
+        "Welcome to Oros! I'm here to help you with all things blockchain and DeFi. Whether you're checking your balances, managing transactions, or exploring earning opportunities, I'll guide you every step of the way. Let's get started. What can I assist you with today?",
+      ),
+    ).toBeVisible();
+  });
+
+  test('receiving a response from the model', async ({ page, context }) => {
+    test.setTimeout(90 * 1000);
+
+    const chat = new Chat(page);
+    await chat.goto();
+
+    await chat.submitMessage(
+      'This is an automated test suite, please respond with the exact text: THIS IS A TEST',
+    );
+
+    await chat.waitForStreamToFinish();
+    await chat.waitForAssistantResponse();
+
+    const messages = await chat.getMessageElementsWithContent();
+    expect(messages.length).toBeGreaterThan(0);
+
+    const attr =
+      await messages[messages.length - 1].getAttribute('data-chat-role');
+    expect(attr).toBe('assistant');
+
+    const responseText = await messages[messages.length - 1].innerText();
+    expect(responseText).toMatch(/THIS IS A TEST/i);
+  });
+
+  describe('wallet tests', () => {
+    let metaMask: MetaMask;
+
+    beforeEach(async ({ context, metaMaskExtensionId }, testInfo) => {
+      expect(metaMaskExtensionId).toBeDefined();
+
+      const walletId = getWalletId(
+        [EvmWalletID.CHAT_TESTING_ACCOUNT_A],
+        testInfo.retry,
+      );
+
+      metaMask = await MetaMask.prepareWallet(
+        context,
+        metaMaskExtensionId,
+        walletId,
+        true,
+      );
+    });
```

Comment:
To help speed up tests, we only register metamask accounts for tests that will use them.